state_machine/drivers/pycubedmini/lib/pycubed.py

```diff
@@ -262,394 +306,168 @@ def reinit(self, dev):
-
-    def new_log(self):
-        """
-        create a new log file
-        """
-        if self.hardware['SDcard']:
-            n = 0
-
-            # iterate through all files in the logs folder
-            for f in listdir('/sd/logs/'):
-                # if the file number is greater than n, set n to file number
-                if int(f[3: -4]) > n:
-                    n = int(f[3: -4])
-
-            # the new log file has number n + 1; n is the current
-            # greatest file number
-            self.logfile = "/sd/logs/log" + "{:03}".format(n + 1) + ".txt"
-
-            # open the new logfile and write the time it was created +
-            # the current status
-            with open(self.logfile, "a") as log:
-                log.write('# Created: {:.0f}\r\n# Status: {}\r\n'.format(
-                    time.monotonic(), self.status))
-
-            # print a confirmation message that a new logfile was created
-            print('New log file:', self.logfile)
-
-    def print_file(self, filedir=None):
-        """
-        print a file given its directory; file directory is by default None
-        """
-
-        # if no file directory is passed, use the directory of the log file
-        if filedir is None:
-            filedir = self.logfile
-
-        print('--- Printing File: {} ---'.format(filedir))
-
-        # open the current file directory as read only, print line by line
-        # (removing whitespace)
-        with open(filedir, "r") as file:
-            for line in file:
-                print(line.strip())
-
-    def send_file(self, c_size, send_buffer, filename):
-        """
-        send a file given character size, buffer size, and the filename
-        """
-
-        # number of packets is the size of the filename / character size
-        num_packets = int(stat(filename)[6] / c_size)
-
-        # open the file
-        with open(filename, "rb") as f:
-            # for each packet
-            for i in range(num_packets + 1):
-                # move the cursor to the end of i * character size,
-                # add to buffer
-                f.seek(i * c_size)
-                f.readinto(send_buffer)
-
-                # return bytes; yield keyword returns without destroying
-                # states of local vars
-                yield bytes([i, 0x45, num_packets])
-
-    def save(self, dataset, savefile=None):
-        """
-        save the passed dataset to the passed savefile
-        dataset should be a set of lists; each line is a list:
-            save(([line1],[line2]))
-        to save a string, make it an item in a list:
-            save(['This is my string'])
-        by default, savefile is not passed
-        """
-        # if no savefile is passed, use the current filename attribute
-        # by default
-        if savefile is None:
-            savefile = self.filename
-
-        # open save file
+
+# initialize Satellite as cubesat
+cubesat = Satellite()
```

Comment:
This seems a bit weird to me. 

I know we've had some debate about whether to structure this as a full class or as a module. @thetazero how are you thinking this structure would be?

Reply:
It seems we are in some kind of middle ground with initialization. But when it comes to accessing the pycubed library we have exactly what I was thinking we were working towards. 

We need some method to initialize everything, and some way for the functions like `gyro` to access the sensors. Currently we are doing this through a Class, and honestly I think this is fine. Perhaps we should rename the cubesat variable to _cubesat however to make it more clear that it is not meant to be accessed by the end user?

Reply:
I really like this idea however unfortunately the class has to be defined before initialization in python.
For example
```python3
x = Test()
print(x)

class Test:
    def __init__(self):
        self.something='something else'
```
Throws the following error:
```
Traceback (most recent call last):
  File "test.py", line 1, in <module>
    x = Test()
NameError: name 'Test' is not defined
```

However a structure like this does work:
```python3
def doSomethingWithX():
    print(f'This is x: {x}')

class Test:
    def __init__(self):
        self.something='something else'

x = Test()
doSomethingWithX()
```

Reply:
fixed! I really like these changes